client/src/components/useFileUpload.ts

```diff
@@ -0,0 +1,202 @@
+import {
+  useQuery,
+  useMutation,
+  useQueryClient,
+  UseQueryResult,
+} from 'react-query'
+import axios, { AxiosRequestConfig } from 'axios'
+import { useState, useRef } from 'react'
+import { IFileInfo, CvrFileType } from './useCSV'
+import { fetchApi, ApiError, addCSRFToken } from '../utils/api'
+
+interface IUseFileUploadProps {
+  url: string
+  formKey: string
+}
+
+export interface IFileUpload {
+  uploadedFile: UseQueryResult<IFileInfo, ApiError>
+  uploadFiles: (files: File[]) => Promise<void>
+  uploadProgress?: number
+  deleteFile: () => Promise<void>
+  downloadFileUrl?: string
+}
+
+export const useUploadedFile = (
+  key: string[],
+  url: string,
+  options: { onFileChange?: () => void; enabled?: boolean } = {}
+) => {
+  const isFirstFetch = useRef(true)
+  const isProcessing = (fileInfo?: IFileInfo) =>
+    fileInfo && fileInfo.processing && !fileInfo.processing.completedAt
+  return useQuery<IFileInfo, ApiError>(key, () => fetchApi(url), {
+    // When a file input dialog closes, it triggers a window focus event,
+    // which causes a refetch by default, so we turn that off.
+    // https://github.com/tannerlinsley/react-query/issues/2960
+    refetchOnWindowFocus: false,
+    refetchInterval: fileInfo => (isProcessing(fileInfo) ? 1000 : false),
+    // Once a file is finished processing or is deleted, call onFileChange
+    // (but don't call it the very first time we load the file)
```

Comment:
Worth surfacing the fact that `onFileChange` isn't called the first time a file is loaded in a top-level comment for the hook, since this is unlike most `onChange` handlers (which don't distinguish `empty state --> populated state` from `populated state --> diff populated state`)

This also makes me wonder whether any consumers will ever need to trigger a change on the initial file upload and whether we should plan for that future

Reply:
Ah! Just meant noting in a function comment so someone doesn't have to read through the implementation of the hook to understand this behavior

Reply:
(Though this may become irrelevant given my idea below)